src/repository/models.py

```diff
@@ -623,6 +632,52 @@ def local_url(self):
 
         return url
 
+    def create_article(self, journal, workflow_stage, journal_license, journal_section, force=False):
+        """
+        Creates an article in a given journal and workflow stage.
+        """
+        if not self.article or force:
+            # create base article
+            article = submission_models.Article.objects.create(
+                journal=journal,
+                owner=self.owner,
+                title=self.title,
+                abstract=self.abstract,
+                license=journal_license,
+                section=journal_section,
+                date_submitted=timezone.now(),
+                comments_editor='Submitted from {}'.format(self.repository.name),
+                stage=workflow_stage,
+            )
+
+            # copy authors to submission
+            for preprint_author in self.preprintauthor_set.all():
+                submission_models.ArticleAuthorOrder.objects.get_or_create(
+                    article=article,
+                    author=preprint_author.account,
+                    defaults={
+                        'order': preprint_author.order
+                    }
+                )
+
+            # snapshot authors
+            article.snapshot_authors()
+
+            # copy preprints latest file and add it as a MS file to the article
+            file = files.copy_preprint_file_to_article(
+                self,
+                article,
+            )
+            article.manuscript_files.add(file)
```

Comment:
I think it makes sense for the function above (`copy_preprint_file_to_article`) to take an argument like `as_manuscript=True` so that the dev doesn't need to remember to add the file to article.manuscripts

Reply:
👍🏻